src/modules/complianceengine/src/lib/Evaluator.cpp

```diff
@@ -306,48 +306,55 @@ Result<Status> Evaluator::EvaluateBuiltinProcedure(const string& procedureName,
     return result.Value();
 }
 
+static constexpr std::size_t cMaxNodeIndicators = 5;
 void NestedListFormatter::FormatNode(const IndicatorsTree::Node& node, std::ostringstream& result, int depth) const
 {
-    for (const auto& child : node.children)
+    for (std::size_t i = 0; i < node.children.size(); i++)
     {
-        for (int i = 0; i < depth; ++i)
+        if (i >= cMaxNodeIndicators)
```

Comment:
max 5 children in addition to max 5 indicators? do we risk cutting off the non-compliant indicators and only showing compliant ones?